net-utils/src/lib.rs

```diff
@@ -392,21 +399,67 @@ pub fn is_host_port(string: String) -> Result<(), String> {
 }
 
 #[derive(Clone, Debug)]
+#[cfg_attr(any(windows, target_os = "ios"), derive(Default))]
 pub struct SocketConfig {
-    pub reuseport: bool,
+    reuseport: bool,
+    #[cfg(not(any(windows, target_os = "ios")))]
+    usage: SocketUsage,
+    #[cfg(not(any(windows, target_os = "ios")))]
+    recv_buffer_size: usize,
+    #[cfg(not(any(windows, target_os = "ios")))]
+    send_buffer_size: usize,
 }
 
+#[cfg(not(any(windows, target_os = "ios")))]
 impl Default for SocketConfig {
```

Comment:
is the idea that we only expose a `SocketConfig` constructor that takes in: `reuseport`, `send_buffer_size`, and `receive_buffer_size`? and the user would have to explicitly set these? 

I worry people would set these incorrectly. and having some default at least falls back on the original settings (using the equivalent buffer sizes previously set using `sysctl`)

Reply:
if we don't have a default socket config, should we also not have `DEFAULT_RECV_BUFFER_SIZE` and `DEFAULT_SEND_BUFFER_SIZE`? just worried people are going to complain that they have to set these manually. And if they have to set them manually, they may just pick some large number (< 128MB) that ends up defeating the whole point of these PRs. 

Also, the kernel doubles the buffer sizes passed in via `setsockopt`. So someone trying to set a buffer size of 128MB is unknowingly allocating 256MB to their sockets per side.

Reply:
ya we could have these `default_ro`, `default_wo`, `default_rw`. and then can have an option to set `reuseport`. we can also have the scaffold for setting buffer sizes directly like:
```
// set read buffer to 16MB and reuseport to true
let socket = SocketConfig::default_ro().recv_buffer_size(16 * 1024 * 1024).reuseport(true); 
```

For this PR we are not changing any of the actual buffer sizes. So, I propose for this PR, we provide `default_wr` that will set buffer sizes to the current 128MB. We leave out `default_ro()` and `default_wo()` on this PR since these defaults don't really make sense if we're not changing the buffer sizes on this PR

And then we at least have the functionality for the next PR where we can add `default_ro()` and `default_wo()` and set the buffer sizes

does that sound good?

Reply:
Works for me; thoughts @t-nelson ?

Reply:
more defaults are the wrong direction IMO. how about this concession..

```rust
#[derive(Default)]
pub struct SocketConfig {
    reuseport: bool,
    recv_buffer_size: Option<usize>,
    send_buffer_size: Option<usize>,
}
```
then when we build the socket, only even call `setsockopt()` when the field is `Some(size)`

Reply:
in this case we just leave the socket buffer size set at whatever the default OS buffer size is? at the end of the day, someone is setting a default. it's either us or the OS. I'm ok with this idea tho

Reply:
yeah we have little control over what the os does. odds are we shouldn't have custom "default" in the first place. the value from the tuning guide was just cribbed out of some blog post and tossed in iirc. no rigor to it. we should be tuning the socket buffers intelligently or not allocating sockets at all

Reply:
> how about this concession..

Works for me